### DISCUSSION TITLE
M1 MBP mysql server was working suddenly stopped

### DISCUSSION THREAD

**Discussion by jdgwf** [Homebrew/discussions#1280] [Everyday usage]
M1 MBP Big Sur 11.2.3

This **was** working until today (the only OS updates I've read have been for Safari Tech Preview).

Service appears to be running according to brew service, but I'm not seeing it in a general process list.

Anywhere else should I check to track this down? 

I'm thinking that it's a false positive of the service being up since there's no corresponding OS process, and I'm not reading anything of value in the Console app.

TIA

`jeff@jeffreys-mbp ~ % mysql mysql 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)
jeff@jeffreys-mbp ~ % brew services list 
Name  Status  User Plist
mysql started root /Library/LaunchDaemons/homebrew.mxcl.mysql.plist
jeff@jeffreys-mbp ~ % mysql mysql       
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)
jeff@jeffreys-mbp ~ % ps aux | grep mysql
jeff              7312   0.0  0.0 408103296   1264 s000  S+   11:30PM   0:00.00 grep mysql
jeff@jeffreys-mbp ~ % ps aux | grep brew 
jeff              7314   0.0  0.0 408112512   1328 s000  S+   11:30PM   0:00.00 grep brew
jeff@jeffreys-mbp ~ % `

**Comment by SMillerDev** [comment]
What happens if you manually start mysqld?

**Reply by jdgwf** [reply]
Strange, it runs and serves just fine - I had to `brew services stop mysql` to actually have the manually run service to open the databases since the InnoDB was locked.

I've unlinked and relinked the services then reattempted to start

got: 
`jeff@jeffreys-mbp ~ % brew services start mysql
==> Successfully started `mysql` (label: homebrew.mxcl.mysql)
jeff@jeffreys-mbp ~ % 
`

and listing services reads its started too:

`jeff@jeffreys-mbp ~ % brew services list
Name  Status  User Plist
mysql started root /Library/LaunchDaemons/homebrew.mxcl.mysql.plist
jeff@jeffreys-mbp ~ % `

while sudo -s'd to root, I'm not seeing any mysql processes running.

Something's definitely going on with the service launcher.

**Reply by jdgwf** [reply]
Got it... turned out that a log file was chowned to root somehow and the permissions to write to it were off:

`jeff@jeffreys-mbp mysql % /usr/local/opt/mysql/bin/mysqld_safe --datadir=/usr/local/var/mysql
2021-04-14T21:01:13.6NZ mysqld_safe Logging to '/usr/local/var/mysql/jeffreys-mbp.jdgwf.local.err'.
/usr/local/opt/mysql/bin/mysqld_safe: line 144: /usr/local/var/mysql/jeffreys-mbp.jdgwf.local.err: Permission denied
2021-04-14T21:01:13.6NZ mysqld_safe Starting mysqld daemon with databases from /usr/local/var/mysql
/usr/local/opt/mysql/bin/mysqld_safe: line 144: /usr/local/var/mysql/jeffreys-mbp.jdgwf.local.err: Permission denied
/usr/local/opt/mysql/bin/mysqld_safe: line 199: /usr/local/var/mysql/jeffreys-mbp.jdgwf.local.err: Permission denied
2021-04-14T21:01:13.6NZ mysqld_safe mysqld from pid file /usr/local/var/mysql/jeffreys-mbp.jdgwf.local.pid ended
/usr/local/opt/mysql/bin/mysqld_safe: line 144: /usr/local/var/mysql/jeffreys-mbp.jdgwf.local.err: Permission denied`

I chown'd that back to my user (the brew user) and it's working again.

Thanks for the help!

This isn't the same script which I launched before - this is the one listed in the .plist args for the service

`/usr/local/opt/mysql/bin/mysqld_safe --datadir=/usr/local/var/mysql`

which I think slowed my diagnosis.
